server/handler/base.go

```diff
@@ -64,7 +66,31 @@ func (b *Base) ProcessPullRequest(ctx context.Context, pullCtx pull.Context, cli
 			return errors.Wrap(err, "unable to determine merge status")
 		}
 		if shouldMerge {
-			bulldozer.MergePR(ctx, pullCtx, merger, config.Merge)
+			failureReason := bulldozer.MergePR(ctx, pullCtx, merger, config.Merge)
+			if failureReason != "" {
+				comments, err := pullCtx.Comments(ctx)
+				if err != nil {
+					logger.Error().Err(err).Msg("failed to list pull request comments")
+				}
+
+				// Only leave a comment if we're presenting new information to the user
+				// Otherwise we're just spamming folks instead of diligently merging PRs
+				shouldComment := true
+				for _, comment := range comments {
+					if strings.EqualFold(comment, failureReason) {
+						shouldComment = false
+					}
+				}
+
+				if shouldComment {
+					_, _, err := client.Issues.CreateComment(ctx, pullCtx.Owner(), pullCtx.Repo(), pullCtx.Number(), &github.IssueComment{
```

Comment:
Does the app already have commenting permissions? I guess it might if we needed write on issues to close them when merging linked PRs. If not, we'll need to update the README.

Reply:
Yeah, the app already asks for write on issues & PRs in order to merge PRs and close linked issues.